plugins/BEdita/Core/src/Model/Table/ObjectsTable.php

```diff
@@ -257,11 +270,56 @@ protected function findType(Query $query, array $options)
      */
     protected function findDateRanges(Query $query, array $options)
     {
-        return $query
-            ->distinct([$this->aliasField($this->getPrimaryKey())])
-            ->innerJoinWith('DateRanges', function (Query $query) use ($options) {
-                return $query->find('dateRanges', $options);
-            });
+        $query = $query->distinct([$this->aliasField($this->getPrimaryKey())]);
+        $join = $this->dateRangesSubQueryJoin($query, $options);
+        if (!empty($join)) {
+            return $join;
+        }
+
+        return $query->innerJoinWith('DateRanges', function (Query $query) use ($options) {
+            return $query->find('dateRanges', $options);
+        });
+    }
+
+    /**
+     * Create a date ranges subquery join if a special sort field is set.
+     *
+     * @param Query $query Query object instance.
+     * @param array $options Array of acceptable date range conditions.
+     * @return Query|null
+     */
+    protected function dateRangesSubQueryJoin(Query $query, array $options): ?Query
+    {
+        $minMaxField = key(
+            array_intersect_key(
+                $options,
+                array_flip(self::DATERANGES_SORT_FIELDS)
+            )
+        );
+        if (empty($minMaxField)) {
+            return null;
+        }
+        if (strpos($minMaxField, 'date_ranges_min_') === 0) {
+            $func = $query->func()->min(substr($minMaxField, 16));
+        } else {
+            $func = $query->func()->max(substr($minMaxField, 16));
+        }
+        unset($options[$minMaxField]);
+        $finder = 'dateRanges';
+        if (empty($options)) {
+            $finder = 'all';
+        }
+        $subQuery = $this->DateRanges->find($finder, $options)
+            ->select([
+                'date_ranges_object_id' => 'object_id',
+                $minMaxField => $func,
```

Comment:
I would simply always add these four fields:
```suggestion
                'date_ranges_min_start_date' => $query->func()->min('start_date'),
                'date_ranges_max_start_date' => $query->func()->max('start_date'),
                'date_ranges_min_end_date' => $query->func()->min('end_date'),
                'date_ranges_max_end_date' => $query->func()->max('end_date'),
```

The load on MySQL shouldn't change by much… these are already indexed fields, and it should already filter by the relevant object IDs at execution time, so I don't expect it to be a big change in query performance, and it makes this function's code much more readable IMHO and easier to maintain.

Reply:
Ok, my only concern was query performance, but we can proceed as you suggest